pkg/ccl/changefeedccl/authorization.go

```diff
@@ -153,6 +153,10 @@ func AuthorizeChangefeedJobAccess(
 		return errors.Newf("could not unwrap details from the payload of job %d", jobID)
 	}
 
+	if len(specs.TargetSpecifications) == 0 {
```

Comment:
could you add a test case for this error branch?

Reply:
It is actually covered by the existing tests which make empty changefeed job payloads, which is how I found it -- you almost never see Insufficient permission errors since we just elide the job from the results, but without this some jobs start showing up in those tests that make empty payloads that shouldn't be there.

This never happens outside of those tests because we never make empty job payloads, so I think it is pretty well exercised, particularly for something we're hoping to deprecate.